heat/core/tensor.py

```diff
@@ -29,17 +30,38 @@ def dtype(self):
     def gshape(self):
         return self.__gshape
 
+    @property
+    def size(self):
+        try:
+            return reduce(lambda i, j: i * j, self.array_shape)
+        except TypeError:
+            # todo: throw a soft warning about taking the number of elements of a single element tensor
+            return 1
+
+    @property
+    def gnumel(self):
+        return self.size
+
+    @property
+    def lnumel(self):
+        return reduce(lambda i, j: i * j, self.lshape)
+
     @property
     def lshape(self):
         if len(self.__array.shape) == len(self.__gshape):
             return tuple(self.__array.shape)
         # edge case when the local data tensor receives no elements after chunking
-        return self.__gshape[:self.__split] + (0,) + self.__gshape[self.split + 1:]
+        return self.__array.shape
+        # return self.__gshape[:self.__split] + (0,) + self.__gshape[self.split + 1:]
 
     @property
     def shape(self):
         return self.__gshape
 
+    @property
+    def array_shape(self):
```

Comment:
how is this different from lshape?

Reply:
artifact from development i missed when cleaning up. now removed